plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/geoip/GeoIpProcessor.java

```diff
@@ -68,13 +71,21 @@ public void execute(Data data) {
             throw new RuntimeException(e);
         }
 
-        final Map<String, Object> geoData;
+        Map<String, Object> geoData;
         switch (dbReader.getMetadata().getDatabaseType()) {
             case "GeoLite2-City":
-                geoData = retrieveCityGeoData(ipAddress);
+                try {
+                    geoData = retrieveCityGeoData(ipAddress);
+                } catch (Exception e) {
+                    geoData = Collections.singletonMap("error", ExceptionsHelper.detailedMessage(e));
```

Comment:
ok, so you suggest that we treat this as kind of a special failure right as it will always tag the document rather than failing? I wonder how many of these special failures we might have in other processors in the future. Is this what logstash currently does? Maybe @talevy can elaborate on this.

ideally I think we should throw exception here, let it bubble and handle the tagging in some central place.